packages/js/plugins/ethereum/src/utils/parsing.ts

```diff
@@ -26,41 +24,23 @@ export function constructAbi(method: string): any[] {
   return abi;
 }
 
-export function parseResult(abi: any[], result: any[] | any): string {
-  if (!(result instanceof Array)) {
-    // if not array, return single value
-    return result.toString();
-  }
-  const isRawAbi = abi[0] instanceof Object;
-  if (!isRawAbi) {
-    return stringifySimpleArray(result);
-  }
-  const outputs = abi[0].outputs;
-  const returnIsStruct = outputs.length > 0 && "components" in outputs[0];
-  if (returnIsStruct) {
-    return stringifyStruct(abi, result);
-  } else {
-    return stringifySimpleArray(result);
-  }
+export function parseResult(result: unknown): string {
+  return JSON.stringify(prepForStringify(result));
 }
 
-function stringifyStruct(abi: any[], result: any): string {
-  const objects: Record<string, string>[] = [];
-  for (const element of result) {
-    const object: Record<string, string> = {};
-    const output = abi[0].outputs[0];
-    for (const component of output.components) {
-      object[component.name] = element[component.name].toString();
+export function prepForStringify(value: unknown): unknown {
+  if (Array.isArray(value)) {
+    const prepared = [];
+    for (const item of value) {
+      prepared.push(prepForStringify(item));
     }
-    objects.push(object);
-  }
-  return JSON.stringify(objects);
-}
-
-function stringifySimpleArray(result: any): string {
-  const objects: string[] = [];
-  for (const element of result) {
-    objects.push(element.toString());
+    return prepared;
+  } else if (
+    typeof value === "object" &&
```

Comment:
What are some examples where this is useful?

Reply:
This is necessary because ethers.js utilizes the `BigNumber` object to represent numbers larger than JS' `number` range (ex: `uint256`). When you simply `JSON.stringify(...)` a `BigNumber` the result is not desirable and just shows the internals of the object. This is because `BigNumber` does not implement a `toJSON()` method. So instead, we leverage object's `toString()` method if it's available.